packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2430,10 +2838,14 @@ export class TransactionController extends BaseControllerV1<
    * @param transactionId - Used to identify original transaction.
    */
   private markNonceDuplicatesDropped(transactionId: string) {
-    const chainId = this.getChainId();
     const transactionMeta = this.getTransaction(transactionId);
+    // NOTE(JL): Should this method be exiting early if getTransaction returns no transaction object?
```

Comment:
yeah it should. i mean there's no tx to work off of.

Reply:
working on this

Reply:
https://github.com/MetaMask/core/pull/3643/commits/90643291820e8eb499439bed6875efdab5df0e6e